src/json-schema/partial/map_state.json

```diff
@@ -49,6 +49,219 @@
+                                            "minLength": 1,
+                                            "examples": [
+                                                "CSV",
+                                                "JSON"
+                                            ]
+                                        },
+                                        "CSVHeaders": {
+                                            "type": "array",
+                                            "minItems": 1,
+                                            "items": {
+                                                "type": "string"
+                                            }
+                                        },
+                                        "CSVHeaderLocation": {
+                                            "type": "string",
+                                            "description": "A string that specifies the location of the CSV column header.",
+                                            "minLength": 1,
+                                            "enum": [
+                                                "FIRST_ROW",
+                                                "GIVEN"
+                                            ]
+                                        },
+                                        "MaxItems": {
+                                            "type": "integer",
+                                            "description": "An integer that limits the number of data items passed to the Map state.",
+                                            "minimum": 0
+                                        },
+                                        "MaxItemsPath": {
+                                            "$ref": "common.json#/definitions/referencePath",
+                                            "description": "A reference path to an integer that limits the number of data items passed to the Map state."
+                                        }
+                                    },
+                                    "required": [
+                                        "InputType"
+                                    ]
+                                },
+                                "Resource": {
+                                    "type": "string",
+                                    "description": "The Amazon S3 API action Step Functions must invoke depending on the specified dataset.",
+                                    "examples": [
+                                        "arn:aws:states:::s3:getObject"
+                                    ]
+                                },
+                                "Parameters": {
+                                    "type": "object",
+                                    "description": "A JSON object that specifies the Amazon S3 bucket name and object key in which the dataset is stored.",
+                                    "properties": {
```

Comment:
`Prefix` is also supported for when the resource is `ListObjectsV2`.

Reply:
Good catch. Looks like we're missing documentation for this in [states-language.net](https://states-language.net/)